### DISCUSSION TITLE
Air Quality Measurement (IAQ Index) with Bosch BME680 sensor

### DISCUSSION THREAD

**Discussion by guiumb** [arendst/Tasmota#13661] [Ideas]
Dear All,

with actual Tasmota release Air Quality Index is not provided becasue sensor send out "only" resistance value related. 
Bosch publish BSEC library that provide a higher-level signal processing.
Could be possible to integrate this code in order to get IAQ Index that is much more "frendly" that resistance value in Ohm ?

This is GitHub library repository:
https://github.com/BoschSensortec/BSEC-Arduino-library

Many thanks in advance
Umberto

**Comment by Vincent1964** [comment]
I understood from the BSEC-Arduino-Library comes as pre-compiled archive library (.a file). I don't think it is possible to add pre-compiled code to the Tasmota project. If anybody knows a way how to integrate pre-compiled code to Tasmota I would give it a try!

**Reply by barbudor** [reply]
You can probably use platformio env to add such library into a build

If the library is not open source, you can integrate it in your own fork but that won't be merged in Tasmota

**Reply by sfromis** [reply]
IIRC, one argument against even trying to integrate that speculative voodoo code ("air quality" is a fictive number) with Tasmota was the large size, nearly 200 KB, way over what was seen as reasonable for a sensor driver. And likely also "more than minor" RAM needs.

**Reply by Jason2866** [reply]
Adding the Bosch closed source *.a file wouldn't be a big thing with Platformio BUT it is only provided for the ESP32 using "xtensa" core. Not the Tasmota way.
